apps/src/blockly/googleBlocklyWrapper.js

```diff
@@ -537,7 +552,17 @@ function initializeBlocklyWrapper(blocklyInstance) {
       }
     },
 
-    createReadOnlyBlockSpace: (container, xml, options) => {
+    // By default we skip loading hidden definitions for a read only block space
+    // because read only block spaces are used for things like hint blocks,
+    // which do not need procedure definitions as they will never be run.
+    // Trying to load hidden definitions when there are none
+    // can wipe our existing hidden definition workspace.
+    createReadOnlyBlockSpace: (
+      container,
+      xml,
+      options,
+      loadHiddenDefinitions = false
```

Comment:
Does this need to be an argument? Theoretically, the workspaces we create with `createReadOnlyBlockSpace` should never have hidden definitions, right?

Reply:
I put it there to be safe but we can probably just use false always.